src/tools/dependencies.ts

```diff
@@ -24,3 +24,25 @@ export function removePackageJSONDependencies(
   const updatedPackageJson = findAndRemoveDependencies(packageJsonRaw, dependenciesToRemove)
   filesystem.write(packageJsonPath, updatedPackageJson)
 }
+
+export function findAndUpdateDependencyVersions(
+  packageJsonRaw: string,
+  dependencies: Record<string, string>,
+): string {
+  let updatedPackageJson = packageJsonRaw
+
+  Object.keys(dependencies).forEach((depName) => {
+    const desiredVersion = dependencies[depName]
+    const regex = new RegExp(`"${depName}"\\s*:\\s*"[^"]+"`, "g")
+    updatedPackageJson = updatedPackageJson.replace(regex, `"${depName}": "${desiredVersion}"`)
```

Comment:
Not important to fix here since we're already doing  this, but this did make me wonder, why are we often operating on the package.json as a string instead of structured data? It seems easier to operate on it as an object for these kinds of changes, to me?

Reply:
I can definitely look into it, that's a good call out, we did that for the `app.json`.

Reply:
Ok so I started to work on this, adding

```
// Record<string, string> could be more complex for dependencies
// For example, we could type them with versions, but also
// GH forks or other custom URLs could be provided there technically, so it gets complex
export interface PackageJson {
  name: string
  version: string
  private: boolean
  main: string
  description: string
  scripts: Record<string, string>
  dependencies: Record<string, string>
  devDependencies: Record<string, string>
  engines: Record<string, string>
}
```

Thoughts on continuing with this? Should I make it a separate PR, it'll impact everything in this range: https://github.com/infinitered/ignite/blob/bc6caf9e4b167a21ae97d0b68aa4dc3f52d5ff7b/src/commands/new.ts#L619-L675

Reply:
Yeah a separate PR seems reasonable as a refactor